modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java

```diff
@@ -1220,8 +1227,8 @@ public RpcMsg eth_newFilter(Object _params) {
         }
 
         FltrLg filter = createFilter(rf);
-        if (filter == null) {
-            return new RpcMsg(null, RpcError.INVALID_PARAMS, "Invalid filter object provided.");
+        if (filter.getFilterError() != null) {
+            return new RpcMsg(null, RpcError.INVALID_PARAMS, "Invalid filter object provided." + filter.getFilterError());
```

Comment:
Minor: a space is needed between the two concatenated messages; same applies to the `if` below